internal/batches/service/service_test.go

```diff
@@ -698,3 +702,268 @@ func TestEnsureDockerImages(t *testing.T) {
 
 	})
 }
+
+func TestService_ParseBatchSpec(t *testing.T) {
+	svc := &Service{}
+
+	tempDir := t.TempDir()
+	tempOutsideDir := t.TempDir()
+	// create temp files/dirs that can be used by the tests
+	_, err := os.Create(filepath.Join(tempDir, "sample.sh"))
+	require.NoError(t, err)
+
+	tests := []struct {
+		name         string
+		batchSpecDir string
+		rawSpec      string
+		expectedSpec *batcheslib.BatchSpec
+		expectedErr  error
+	}{
+		{
+			name: "simple spec",
+			rawSpec: `
+name: test-spec
+description: A test spec
+`,
+			expectedSpec: &batcheslib.BatchSpec{Name: "test-spec", Description: "A test spec"},
+		},
+		{
+			name: "unknown field",
+			rawSpec: `
+name: test-spec
+description: A test spec
+some-new-field: Foo bar
+`,
+			expectedErr: errors.New("parsing batch spec: Additional property some-new-field is not allowed"),
+		},
+		{
+			name:         "mount absolute file",
+			batchSpecDir: tempDir,
+			rawSpec: fmt.Sprintf(`
+name: test-spec
+description: A test spec
+steps:
+  - run: /tmp/sample.sh
+    container: alpine:3
+    mount:
+      - path: %s
+        mountpoint: /tmp/sample.sh
```

Comment:
Could it be worth adding a test case with multiple steps/multiple mounted files as well?